### DISCUSSION TITLE
x-on is not re-rendering the template

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#1017] [1. Help] [answered]
```
<script>let contacts = [];</script>
<div x-data="{ persons: contacts }">
    <template x-for=" person in persons ">
        <div x-text=" person "></div>
    </template>            
    <button x-on:click="contacts.push('Jack'); persons = contacts; console.log(persons.length);">Click</button>
</div>
```

**Comment by HugoDF** [comment, marked as answer]
`contacts` is not used by Alpine once the component has initialised. `x-data` is run/evaluated and wrapped in a proxy, at that point `persons` is within Alpine scope and reactive, `contacts` is not.

In order to get a re-render you'll want to push to `persons` not `contacts` (`contacts` is a global and is not wrapped or watched by Alpine).

Side note: `.push` isn't always reliable so you might want to re-assign `persons = persons.concat('Jack')`.
